src/commands/functions/functions-create.js

```diff
@@ -93,13 +93,17 @@ const filterRegistry = function (registry, input) {
     })
 }
 
-const formatRegistryArrayForInquirer = function (lang) {
-  const folderNames = fs.readdirSync(path.join(templatesDir, lang))
+const formatRegistryArrayForInquirer = function (lang, isEdgeFunc) {
+  const functionType = isEdgeFunc ? 'edge' : 'serverless'
+  const folderNames = fs.readdirSync(path.join(templatesDir, lang, functionType))
   const registry = folderNames
     // filter out markdown files
     .filter((folderName) => !folderName.endsWith('.md'))
-    // eslint-disable-next-line n/global-require, import/no-dynamic-require
-    .map((folderName) => require(path.join(templatesDir, lang, folderName, '.netlify-function-template.js')))
+
+    .map((folderName) =>
+      // eslint-disable-next-line n/global-require, import/no-dynamic-require
+      require(path.join(templatesDir, lang, functionType, folderName, '.netlify-function-template.js')),
```

Comment:
I wonder if you've considered keeping the templates organised by language and using the existing `.netlify-function-template.js` file to store the type (i.e. `serverless` or `edge`)? Feels like it could simplify the logic a little bit, because you'd still load files from the same location but rather than filtering templates by `language` you'd also look at `type`?

Reply:
Hmm... I'll have to think about this one a little more because I don't quite have a clear understanding of how the `.netlify-function-template.js` files work. It would definitely be nice not to change the entire directory structure, if there's a simpler way.

Reply:
I think that file is just a sort of manifest file that tells CLI some information about the template. We can extend it with any data we want.